core/src/main/java/com/amplifyframework/hub/AWSHubPlugin.java

```diff
@@ -68,8 +65,7 @@
                         !subscription.getHubEventFilter().filter(hubEvent)) {
                     continue;
                 }
-
-                mainHandler.post(() -> subscription.getHubSubscriber().onEvent(hubEvent));
```

Comment:
This line is problematic when writing a unit test that exercises the `AWSHubPlugin`. It is also a strange choice, to callback on the main thread, when the subscription did not necessarily happen on the main thread. So, while this line doesn't appear to have to do with the purpose of the PR at first glance -- it's towards making this class test-able.